layers/core_checks/cc_pipeline_graphics.cpp

```diff
@@ -4238,31 +4150,108 @@ bool CoreChecks::ValidateDrawPipelineVertexBinding(const vvl::CommandBuffer &cb_
+                             print_binding(binding_description).c_str());
             continue;
         }
 
-        if (vertex_buffer_binding.second.buffer == VK_NULL_HANDLE) {
+        // This means the app actively set the buffer to null
+        // Going to hit VUID-vkCmdBindVertexBuffers-pBuffers-04001 first anyway
+        if (vertex_buffer_binding->buffer == VK_NULL_HANDLE) {
             if (!enabled_features.nullDescriptor) {
                 const LogObjectList objlist(cb_state.Handle(), pipeline.Handle());
                 skip |= LogError(vuid.vertex_binding_null_04008, objlist, vuid.loc(),
-                                 "Vertex binding %" PRIu32
-                                 " is VK_NULL_HANDLE. (Most likely you forgot to call vkCmdBindVertexBuffers)",
-                                 vertex_buffer_binding.first);
+                                 "%s has a buffer of VK_NULL_HANDLE bound, but nullDescriptor is not enabled.",
+                                 print_binding(binding_description).c_str());
             }
-        } else if (!Get<vvl::Buffer>(vertex_buffer_binding.second.buffer)) {
+            continue;
+        }
+
+        const auto vertex_buffer_state = Get<vvl::Buffer>(vertex_buffer_binding->buffer);
+        if (!vertex_buffer_state) {
             const LogObjectList objlist(cb_state.Handle(), pipeline.Handle());
             skip |= LogError(vuid.vertex_binding_04007, objlist, vuid.loc(),
-                             "Vertex binding %" PRIu32 " is not a valid VkBuffer. (Check the buffer set in vkCmdBindVertexBuffers)",
-                             vertex_buffer_binding.first);
+                             "%s has an invalid/destroyed buffer bound with vkCmdBindVertexBuffers.",
+                             print_binding(binding_description).c_str());
+            continue;
         }
-    }
 
-    skip |= ValidateDrawPipelineVertexAttribute(cb_state, pipeline, vuid);
+        for (const auto &location : binding_description.locations) {
+            const auto attr_index = location.second.index;
+            const auto &attr_desc = location.second.desc;
+
+            const VkDeviceSize attribute_offset = attr_desc.offset;
+            const VkFormat attribute_format = attr_desc.format;
+            const VkDeviceSize vertex_buffer_stride = vertex_buffer_binding->stride;
+            if (pipeline.IsDynamic(CB_DYNAMIC_STATE_VERTEX_INPUT_BINDING_STRIDE)) {
+                const VkDeviceSize attribute_binding_extent = attribute_offset + GetVertexInputFormatSize(attribute_format);
+                if (vertex_buffer_stride != 0 && vertex_buffer_stride < attribute_binding_extent) {
+                    const LogObjectList objlist(cb_state.Handle(), pipeline.Handle());
+                    skip |= LogError("VUID-vkCmdBindVertexBuffers2-pStrides-06209", objlist, vuid.loc(),
+                                     "(binding %" PRIu32 "location %" PRIu32 ") The pStrides value (%" PRIu64
```

Comment:
```suggestion
                                     "(attribute binding %" PRIu32 ", attribute location %" PRIu32 ") The pStrides value (%" PRIu64
```
(not immediately clear what "binding" refers to, so maybe add "attribute")

Reply:
agree, this was just copied over